hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -1765,16 +1765,7 @@ public OzoneKeyDetails getS3KeyDetails(String bucketName, String keyName)
   @Override
   public OzoneKeyDetails getS3KeyDetails(String bucketName, String keyName,
                                          int partNumber) throws IOException {
-    OmKeyInfo keyInfo = getS3KeyInfo(bucketName, keyName, false);
-    List<OmKeyLocationInfo> filteredKeyLocationInfo = keyInfo
-        .getLatestVersionLocations().getBlocksLatestVersionOnly().stream()
-        .filter(omKeyLocationInfo -> omKeyLocationInfo.getPartNumber() ==
-            partNumber)
-        .collect(Collectors.toList());
-    keyInfo.updateLocationInfoList(filteredKeyLocationInfo, false);
-    keyInfo.setDataSize(filteredKeyLocationInfo.stream()
-        .mapToLong(OmKeyLocationInfo::getLength)
-        .sum());
+    OmKeyInfo keyInfo = getS3PartKeyInfo(bucketName, keyName, partNumber);
     return getOzoneKeyDetails(keyInfo);
```

Comment:
Please generate a new `OzoneManagerVersion` decide on whether to use old or new client implementation based on the OM version. See `listStatusLight`.

Reply:
Another way is to just use `getS3PartkeyInfo` (only if we are using the `multipartNumber` instead of the new `partNumber`, see my other comments), but still retain the client-side filtering logic. That way, it should be compatible.

Reply:
@ivandika3 I'm not sure that I fully understood your advice. Logic must moved to the server side. The second variant is not acceptable. Do you offer to create a new method with server-side filtering? Also, need to create the same request/response model but with a different name? I suspect that the reason is backward compatibility. But does current realization break something? Is it possible, when the client was updated and the server-side part was not?

Reply:
Thanks for your explanation. Now it is absolutely clear.